### PR TITLE
chore: add pull request title check

### LINKED ISSUE #4891: Considering enforcing conventional commit specification
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

Whilst the release notes are great at showing what changed and who committed what, the individual items are a mixed bag, and make it difficult to skim read.

As an example from v4.19.0:
image

From a glance, I can't tell what that `pluginName will be exposed in FastifyInstance` change is. Is it a bugfix? Is it a new feature? Was it just the documentation that was updated? The other two items in the list I can clearly see what type of change they were.

I think this could be remedied by enforcing the use of the [conventional commit specification](https://www.conventionalcommits.org/en/v1.0.0/) using the [Angular convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type) with commitlint in a husky hook and/or the [commitlint GH Action](https://github.com/wagoid/commitlint-github-action).

### PR DESCRIPTION
**Pull request by Uzlopak**
Resolves #4891 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: .github/workflows/pull-request-title.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+
+jobs:
+  pull-request-title-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: fastify/action-pr-title@v0
+      with:
+        prefixes: 'feature:,chore:,fix:,docs:'
```
Comment (Fdawgs) [inline .github/workflows/pull-request-title.yml:12]:
```suggestion
        prefixes: 'build:,chore:,ci:,docs:,feat:,fix:,perf:,refactor:,style:,test:'
```
Adds all the conventional commit prefixes. 😊

**Review by Fdawgs — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-request-title.yml (hunk 1 above)
Comment (Uzlopak) [reply .github/workflows/pull-request-title.yml]: Gonna catch'em all.

**Review by Fdawgs — APPROVED** [review]

**Comment by Uzlopak** [comment]
lets merge and check

**Review by Fdawgs — CHANGES_REQUESTED** [review]
Sorry blocking again, need to update the prefixes as users can specify a scope i.e. `chore(scope here):`.

Will revisit on my lunch break!

Edit: seems I commented just as it was merged! Will open another PR for this.

**Comment by Uzlopak** [comment]
LOL. Too late. 

We also forgot "types:"

**Comment by Uzlopak** [comment]
@Fdawgs 
I tested it so far. It seems to work as expected. 

Tbh, i dont want to write the regex :D. Waiting for our PR ;)

**Pull request merged**